data/api/client-server/definitions/client_event_without_room_id.yaml

```diff
@@ -112,3 +113,22 @@ properties:
             this.
         title: EventContent
         type: object
+      membership:
+        description: |
+          The room membership of the user making the request, at the time of the event.
+
+          This property is the value of the `membership` property of the
+          requesting user's [`m.room.member`](/client-server-api#mroommember)
+          state at the point of the event, including any changes caused by the
+          event. If the user had yet to join the room at the time of the event
+          (i.e, they have no `m.room.member` state), this property is set to
+          `leave`.
+
+          It is recommended that homeservers SHOULD populate this property
+          wherever practical, but they MAY omit it if necessary (for example,
+          if calculating the value is expensive, servers might choose to only
+          implement it in encrypted rooms). The property is *not* normally populated
+          in the application service API (where there is no clear definition of "requesting user").
```

Comment:
I couldn't find this bit about application services in the MSC -- I would think you'd populate it with whoever the AS is acting as?

Reply:
I was thinking the "virtual user" provided by the `user_id` query parameter: https://spec.matrix.org/v1.10/application-service-api/#identity-assertion

But now we're kind of making stuff up that's not in the MSC. So probably best to leave it and if someone else wants it for AS they can define that in a new MSC.

Reply:
```suggestion
          implement it in encrypted rooms). The property is *not* normally populated
          in events pushed to application services via the application service transaction API
          (where there is no clear definition of "requesting user").
```

Reply:
Oh yes, that makes sense!

Reply:
That sounds good, can you add a cross link to the AS docs?

Reply:
yeees, but then it looks even more ridiculous in the AS docs itself

Reply:
What do you mean that it looks ridiculous? I'm ok with it as is, if you think it looks best now.

Reply:
> What do you mean that it looks ridiculous? 

Currently, under https://pr1847--matrix-spec-previews.netlify.app/application-service-api/#put_matrixappv1transactionstxnid, we have:

![image](https://github.com/matrix-org/matrix-spec/assets/1389908/ea66fcde-2f5b-469a-8afe-1d340bd4dc83)

That looks moderately ridiculous imho (why is it listed at all?), but it's hard to do much about. OTOH, if it links straight back to https://pr1847--matrix-spec-previews.netlify.app/application-service-api/#put_matrixappv1transactionstxnid, it might be even sillier.

Or maybe we should accept the awkwardness, and at least linking it would be clear?

Reply:
Ah -- the same bit gets added in both endpoints. I understand now. 👍 So yes -- I agree that would be a silly link to include.

I think ideally it wouldn't even be listed on the AS spec, but I have no idea how much refactoring that would need.